conformance/conformance_php.php

```diff
@@ -27,57 +36,89 @@
+          $response->setSkipped("PHP doesn't support editions-specific features yet");
           return $response;
-        }
-      } elseif ($request->getMessageType() == "protobuf_test_messages.proto2.TestAllTypesProto2") {
-        $response->setSkipped("PHP doesn't support proto2");
+        case "":
+          trigger_error("Protobuf request doesn't have specific payload type", E_USER_ERROR);
+        default:
+          trigger_error(
+            sprintf("Protobuf request doesn't support %s message type", $request->getMessageType()),
+            E_USER_ERROR
+          );
+      }
+
+      try {
+        $test_message->mergeFromString($request->getProtobufPayload());
+      } catch (Exception $e) {
+        $response->setParseError($e->getMessage());
         return $response;
-      } else {
-        trigger_error("Protobuf request doesn't have specific payload type", E_USER_ERROR);
       }
-    } elseif ($request->getPayload() == "json_payload") {
+      break;
+    case "json_payload":
+      switch ($request->getMessageType()) {
+        case "protobuf_test_messages.editions.proto3.TestAllTypesProto3":
+          $test_message = new \Protobuf_test_messages\Editions\Proto3\TestAllTypesProto3();
+          break;
+        case "protobuf_test_messages.editions.proto2.TestAllTypesProto2":
+          $response->setSkipped("PHP doesn't support proto2");
+          return $response;
+        default:
+          $test_message = new \Protobuf_test_messages\Proto3\TestAllTypesProto3();
+      }
+
       $ignore_json_unknown =
           ($request->getTestCategory() ==
               TestCategory::JSON_IGNORE_UNKNOWN_PARSING_TEST);
       try {
           $test_message->mergeFromJsonString($request->getJsonPayload(),
-                                             $ignore_json_unknown);
+                                            $ignore_json_unknown);
       } catch (Exception $e) {
           $response->setParseError($e->getMessage());
           return $response;
       }
-	} elseif ($request->getPayload() == "text_payload") {
-		$response->setSkipped("PHP doesn't support text format yet");
-        return $response;
-	} else {
+      break;
+	  case "text_payload":
+		  $response->setSkipped("PHP doesn't support text format yet");
+      return $response;
+	  default:
       trigger_error("Request didn't have payload.", E_USER_ERROR);
-    }
+  }
 
-    if ($request->getRequestedOutputFormat() == WireFormat::UNSPECIFIED) {
+  switch ($request->getRequestedOutputFormat()) {
+    case WireFormat::UNSPECIFIED:
       trigger_error("Unspecified output format.", E_USER_ERROR);
-    } elseif ($request->getRequestedOutputFormat() == WireFormat::PROTOBUF) {
+    case WireFormat::PROTOBUF:
       $response->setProtobufPayload($test_message->serializeToString());
-    } elseif ($request->getRequestedOutputFormat() == WireFormat::JSON) {
+      break;
+    case WireFormat::JSON:
```

Comment:
Are you missing a a TEXT case?

Reply:
Added explicit ignoring of text output cases in [6ccd522](https://github.com/protocolbuffers/protobuf/pull/16712/commits/6ccd522c16415aa55d6b094582b95c0866854235)